CenterForOpenScience/osf.io
# [QA Prep Required][Migration Prep Required] Feature/Registrations Files Count [ENG-165]

## Purpose

Adds file counts to registrations - caches counts on model since this value isn't going to change.
Adds a data migration to populate `files_count` field for existing registrations.

## Changes

Reticketing here so PR is clean - my commit includes a merge commit. My commits here are speeding up the migration. [I only changed the migration file]

follow up for https://github.com/CenterForOpenScience/osf.io/pull/8915
 
## QA Notes
Pre-migration, create projects with an assortment of files and folders, and then create a registration from this project.  Note what the file count should be ahead of time. Post-migration, verify that file counts displayed match pre-migration counts.  Create new registrations post-migration and verify that the file count is correct.

## Documentation



## Side Effects



## Ticket

https://openscience.atlassian.net/browse/ENG-165
